pkg/kapp/cmd/appchange/list.go

```diff
@@ -75,14 +110,18 @@ func (t AppChangesTable) Print(ui ui.UI) {
 			uitable.NewHeader("Description"),
 			nsHeader,
 		},
-
 		SortBy: []uitable.ColumnSort{
 			{Column: 1, Asc: false},
 			{Column: 0, Asc: true}, // in case start time are same
 		},
 	}
 
 	for _, change := range t.Changes {
+		if (!t.TimeFlags.BeforeTime.IsZero() && !change.Meta().StartedAt.Before(t.TimeFlags.BeforeTime)) ||
+			(!t.TimeFlags.AfterTime.IsZero() && !change.Meta().StartedAt.After(t.TimeFlags.AfterTime)) {
```

Comment:
*question*: Do we need this `t.TimeFlags.AfterTime.IsZero()`?
I am thinking that if `AfterTime` is `Zero`, then `change.Meta().StartedAt.After(t.TimeFlags.AfterTime)` would always be false. Or maybe I am missing some edge case!

Reply:
Also, would it make sense to have the `!` outside, in my mind that would be more readable.
```suggestion
			!(t.TimeFlags.AfterTime.IsZero() && change.Meta().StartedAt.After(t.TimeFlags.AfterTime)) {
```

Reply:
@praveenrewar this would change the condition....Please check. However, if the condition can be made simpler, I am all for it

Reply:
!(t.TimeFlags.AfterTime.IsZero() || change.Meta().StartedAt.After(t.TimeFlags.AfterTime))

Reply:
My bad. @renuy Thank for pointing it out.

Reply:
I think we can move the append statement inside a simplified if condition instead of using continue.

Reply:
> I think we can move the append statement inside a simplified if condition instead of using continue.

I have used `continue` here as we just have to skip which are not satisfying the filter criteria and append everything else. Also looks like if we start appending inside some `if condition` then we also have to handle a scenario when no filter is applied append all of them.

Reply:
Also up for discussing it on call